src/browser-detection/BrowserDetection.ts

```diff
@@ -147,45 +155,41 @@ class BrowserDetection {
      */
     static isPrivateMode() {
         return new Promise((resolve) => {
-            const on = () => { resolve(true) }; // is in private mode
-            const off = () => { resolve(false) }; // not private mode
+            const on = () => resolve(true); // is in private mode
+            const off = () => resolve(false); // not private mode
             // using browser detection by feature detection here, also see https://stackoverflow.com/a/9851769
             // These seem to be partly outdated though. Might want to consider using user agent based detection.
-            const isSafari = () => {
-                return (
-                    /Constructor/.test(window.HTMLElement) ||
-                    (function (root) {
-                            return (!root || root.pushNotification).toString() === '[object SafariRemoteNotification]';
-                        }
-                    )(window.safari)
-                );
-            };
+            const isSafari = () => /Constructor/.test(window.HTMLElement)
+                || (window.safari && window.safari.pushNotification).toString() === '[object SafariRemoteNotification]';
```

Comment:
You need to use `!!window.safari` to convert `undefined` to a boolean from which a string can be build.
But better would be to change this to:
```suggestion
                || (
                    window.safari
                    && window.safari.pushNotification
                    && window.safari.pushNotification.toString() === '[object SafariRemoteNotification]'
                );
```